services/transfer_service.go

```diff
@@ -40,32 +48,38 @@ func writeError(w http.ResponseWriter, message string, code int) {
 	w.Write(data)
 }
 
-// a response for an ElasticSearch query (GET)
-type ElasticSearchResponse struct {
+// a response for a database-related query (GET)
+type DatabaseResponse struct {
+	Id           string `json:"id" example:"jdp" `
+	Name         string `json:"name" example:"JGI Data portal"`
+	Organization string `json:"organization" example:"Joint Genome Institute"`
+	URL          string `json:"url" example:"https://data.jgi.doe.gov"`
+}
+
+// a response for a query (GET)
+type SearchResultsResponse struct {
 	// name of organization database
-	Database string `json:"database"`
+	Database string `json:"database" example:"jdp" doc:"the database searched"`
 	// ElasticSearch query string
-	Query string `json:"query"`
+	Query string `json:"query" example:"prochlorococcus" doc:"the given query string"`
 	// Resources matching the query
-	Resources []frictionless.DataResource `json:"resources"`
+	Resources []frictionless.DataResource `json:"resources" doc:"an array of Frictionless DataResources"`
 }
 
 // a request for a file transfer (POST)
 type TransferRequest struct {
 	// name of source database
-	Source string `json:"source"`
+	Source string `json:"source" example:"jdp" doc:"source database identifier"`
 	// identifiers for files to be transferred
-	FileIds []string `json:"file_ids"`
+	FileIds []string `json:"file_ids" doc:"source-specific identifiers for files to be transferred"`
```

Comment:
is it possible to add `example`s for data structures like an array?

Reply:
I think we could do this with a tag like `example:"[\"fileid1\", \"fileid2\"]"`, but I haven't tried it.

Reply:
Just reading the documentation for Huma and it looks like there are all kinds of fancy things possible -- as long as you believe!

Reply:
I'd love to know what you think of Huma, though I know you're new to Golang. I distrust magical things that do everything for you, but Huma seems to hike right up to where it might start violating the Principle of Least Astonishment and then stops. :-)

Reply:
It looked pretty nice from the docs--similar to a couple of other frameworks I've used--and I like the JSONschema integration. You need some nice edge cases to test it with, though -- it's easy for the docs to talk up all the fancy features but neglect to mention that there's no support for <critical feature x> or anything that's slightly outside what the original dev had imagined.

Reply:
I'll try to stick in some more complicated doc examples like arrays and objects before we merge this. Let me know if there's anything specific you'd like me to try.

Reply:
The array example you mentioned does work as expected. Objects seem to be handled by field doc tags. So for now I'd say we're probably safe going with Huma. I'm sure we'll encounter some interesting edge cases as we flesh things out!